lib/wallet/src/Cardano/Wallet/Types/Read/Eras.hs

```diff
@@ -0,0 +1,94 @@
+
+-- |
+-- Copyright: © 2020-2022 IOHK
+-- License: Apache-2.0
+--
+-- Re-export `EraValue` library.
+--
+
+module Cardano.Wallet.Types.Read.Eras
+  ( -- * eras
+    KnownEras
+  , knownEraIndices
+    -- * era bounded values
+  , EraValue
+  , eraValueSerialize
+  , eraValueS
+  , fromEraValueS
+  , toEraValueS
+  , extractEraValue
+  -- * era specific prisms
+  , MkEraValue (..)
+  , byron
+  , shelley
+  , allegra
+  , mary
+  , alonzo
+  , babbage
+  -- * era specific prism shortcuts
+  , inject
+  , eject
+  -- * sum type encoding
+  , EraValueS (..)
+  -- * specials
+  , sequenceEraValue
+  -- * era bounded functions
+  , EraFun
+  , eraFunR
+  , EraFunR(..)
+  , fromEraFunR
+  , toEraFunR
+  -- * composing era functions
+  , (*.*)
+  , (*.**)
+  , (*&&&*)
```

Comment:
Alas, adding new operators has a large cognitive cost and should be done only if the benefit is larger.

For example, as ` Control.Category` uses the `PolyKinds` extensions, we can actually make an instance

```
instance Category EraFun
```

giving us the genuine `(.)` operator.

If we really need an operator symbol, then a good principle is to model it on an existing operator. Knowing `(&&&)`, I can kind of understand what `(*&&&*)` might do, but I'm entirely lost on `(.*)`.

This pull request doesn't contain any use cases for `(*&&&*)` and `(*.**)` — but you probably want to use them later. 🤔 This makes them hard to review upfront. I suggest the following: We add functions with names

```
pair :: EraFun f g -> EraFun f h -> EraFun f (g :*: h)
mapRight :: EraFun g h -> EraFun f (Either e :.: g) -> EraFun f (Either e :.: h)
```

and promote them to operators only after they have proven their worth. Here, the type of `mapRight` is a bit more specialized than the operator, but I believe that this is also the predominant use case.

Reply:
So, great if we can reuse (.) , approved, thanks for checking it.

I'm against function pair (I switched from there) here because of inflixr , we want to use infixr  as (:*:) has infixr so we do not mess up with order.

mapRight is not .... Right . It's inside, or below. This I will be more for changing with a function I think.